src/wallet/wallet.cpp

```diff
@@ -2653,11 +2653,11 @@ static uint32_t GetLocktimeForNewTransaction(interfaces::Chain& chain, const uin
     return locktime;
 }
 
-OutputType CWallet::TransactionChangeType(OutputType change_type, const std::vector<CRecipient>& vecSend)
+OutputType CWallet::TransactionChangeType(const Optional<OutputType>& change_type, const std::vector<CRecipient>& vecSend)
 {
     // If -changetype is specified, always use that change type.
-    if (change_type != OutputType::CHANGE_AUTO) {
-        return change_type;
+    if (change_type) {
+        return *change_type;
```

Comment:
I guess it's non-trivial to expand our c++17 `std::optional` substitute with [std::optional<T>::value](https://en.cppreference.com/w/cpp/utility/optional/value)?

Reply:
Sorry I don't follow. https://en.cppreference.com/w/cpp/utility/optional/operator* exists in C++17

Reply:
So you can use `change_type->value()` which looks prettier imo, and - more importantly - raises an exception if `change_type` isn't set.

Reply:
change_type can never be unset in this line because the line is preceded by a check to see if it is set, so I'd rather not annotate this function to throw an exception when in reality it can not throw an exception.